frontend/src/screens/internal-apps/BuzzPay.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import AppHeader from "src/components/AppHeader";
+import AppCard from "src/components/connections/AppCard";
+import Loading from "src/components/Loading";
+import { ExternalLinkButton } from "src/components/ui/button";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { useToast } from "src/components/ui/use-toast";
+import { useApps } from "src/hooks/useApps";
+import {
+  App,
+  AppPermissions,
+  CreateAppRequest,
+  CreateAppResponse,
+  UpdateAppRequest,
+} from "src/types";
+import { handleRequestError } from "src/utils/handleRequestError";
+import { request } from "src/utils/request";
+
+export function BuzzPay() {
+  const { data: apps, mutate: reloadApps } = useApps();
+  const [creatingApp, setCreatingApp] = React.useState(false);
+  const { toast } = useToast();
+
+  if (!apps) {
+    return <Loading />;
+  }
+  const app = apps.find(
+    (app) =>
+      app.metadata?.app_store_app_id === "buzzpay" &&
+      app.metadata.connection_secret
+  );
+
+  function createApp() {
+    setCreatingApp(true);
+    (async () => {
+      try {
+        const name = "BuzzPay";
+        if (apps?.some((existingApp) => existingApp.name === name)) {
+          throw new Error("A connection with the same name already exists.");
+        }
+
+        const createAppRequest: CreateAppRequest = {
+          name,
+          scopes: ["get_info", "lookup_invoice", "make_invoice"],
+          isolated: false,
+          metadata: {
+            app_store_app_id: "buzzpay",
+          },
+        };
+
+        const createAppResponse = await request<CreateAppResponse>(
+          "/api/apps",
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify(createAppRequest),
+          }
+        );
+
+        if (!createAppResponse) {
+          throw new Error("no create app response received");
+        }
+
+        const app = await request<App>(
+          `/api/apps/${createAppResponse.pairingPublicKey}`
+        );
```

Comment:
we can abstract this away a bit into a `createApp` `updateApp` and `fetchApp` function maybe. 
in the future it makes it easier for such apps to be build

Reply:
done - see `requests/createApp`. Updating was removed from this PR, and fetching is a one-liner using SWR